developer/contributing.rst

```diff
@@ -1,2 +1,48 @@
 Contributing guidelines
-=======================
\ No newline at end of file
+=======================
+
+We are glad that you choose this organization to give your time to and contribute. Thank you for your interest incontributing to OpenWISP.
+
+Settings things up
+------------------
+
+Please follow up the instructions in the README of the repository to which you want to contribute to, so that you can set up the project.
+
+Finding Something to Work
+-------------------------
+
+Visit any repository and then to try to explore the project's codebase. Use it and give us your opinion about any change that might be helpful to you as well as to us. If you find any bug or error or something that is worth mentioning please create an **issue** regarding that bug or error.
+
+If you are unable to find anything on your own to be worked upon, don't worry we got your back. Visit that project's **Issues** tab and explore those issues that might interests you, comment on the issue thread and we'll help you get along with that.
+
+If neither of the above works for you please post your views on our `gitter channel. <https://gitter.im/openwisp/general>`_
+
+Steps to Submit your Code or Changes
+------------------------------------
+
+Our main development branch is master it's our central development branch. Your code should be pushed from a different branch and then it is merged into master after proper review from the project maintainers.
+
+1. Create a new branch for your development and a self-descriptive branch name:
+
+     git pull origin master
```

Comment:
make sure these commands appear on different lines

Reply:
Yeah. Doing it